client/src/cmdlfem4x50.c

```diff
@@ -0,0 +1,720 @@
+    strcat(string, pstring);
+
+    PrintAndLogEx(NORMAL,string);
+   
+    string[0] = '\0';
+}
+
+static void print_result(const em4x50_word_t *words,  int fwr,  int lwr) {
+    
+    // print available information for given word from fwr to lwr, i.e.
+    // bit table + summary lines with hex notation of word (msb + lsb)
+    
+    char string[400] = {0};
+    char pstring[100] = {0};
+
+    for (int i = fwr; i <= lwr; i++) {
+
+        // blank line before each bit table
+        PrintAndLogEx(NORMAL, "");
+
+        // print bit table
+        print_bit_table(words[i]);
+         
+        // final result
+        sprintf(pstring, "\n  word[%i] msb: " _BLUE_("0x"), i);
+        strcat(string, pstring);
+
+        for (int j = 0; j < 4; j++) {
+            sprintf(pstring, _BLUE_("%02x"), words[i].byte[j]);
+            strcat(string, pstring);
+        }
+        
+        sprintf(pstring, "\n  word[%i] lsb: " _BLUE_("0x"), i);
+        strcat(string, pstring);
+
+        for (int j = 0; j < 4; j++) {
+            sprintf(pstring, _BLUE_("%02x"), msb2lsb(words[i].byte[3-j]));
+            strcat(string, pstring);
+        }
+        
+        PrintAndLogEx(NORMAL,string);
+
+        string[0] = '\0';
+    }
+}
+
+static void print_info_result(PacketResponseNG *resp, const em4x50_data_t *etd, bool bverbose) {
+
+    // display all information info result in structured format
+    
+    bool bpwd_given = etd->pwd_given;
+    bool bsuccess = (bool)resp->oldarg[0];
+    bool blogin = (bool)resp->oldarg[1];
+    bool bpwc, braw;
+    int fwr, lwr, fwrp, lwrp, fwwi, lwwi;
+    uint8_t *data = resp->data.asBytes;
+    em4x50_word_t words[34];
+    char pstring[100] = {0};
+    char string[200] = {0};
+
+    prepare_result(data, 0, 33, words);
+
+    bpwc = check_bit_in_byte(7, words[2].byte[2]);   // password check
+    braw = check_bit_in_byte(6, words[2].byte[2]);   // read after write
+    fwr = msb2lsb(words[2].byte[0]);    // first word read
+    lwr = msb2lsb(words[2].byte[1]);    // last word read
+    fwrp = msb2lsb(words[1].byte[0]);   // first word read protected
+    lwrp = msb2lsb(words[1].byte[1]);   // last word read protected
+    fwwi = msb2lsb(words[1].byte[2]);   // first word write inhibited
+    lwwi = msb2lsb(words[1].byte[3]);   // last word write inhibited
+    
+    // data section
+    PrintAndLogEx(NORMAL, _YELLOW_("\n  em4x50 data:"));
+
+    if (bverbose) {
+
+        // detailed data section
+        print_result(words, 0, 33);
+        
+    } else {
+
+        // condensed data section
+        for (int i = 0; i <= 33; i++) {
+     
+            sprintf(pstring, "  word[%2i]:  ", i);
+            strcat(string, pstring);
+
+            for (int j = 0; j < 4; j++) {
+                sprintf(pstring, "%02x", words[i].byte[j]);
+                strcat(string, pstring);
+            }
+            
+            switch(i) {
+                case 0:
+                    sprintf(pstring, _BLUE_("  password, write only"));
```

Comment:
BLUE is reserved for system.
Recommend using  YELLOW or GREEN,

Reply:
done